File: pset6/caesar.py
import cs50
import sys

# check for proper usage
if len(sys.argv) != 2:
    print('Usage: caesar.py key')
    exit(1)

# convert first argument (key) to integer
key = int(sys.argv[1])

# get plaintext
print('plaintext: ', end = '')
plain = cs50.get_string()

print('ciphertext: ', end = '')
# check for each letter in plaintext if it's alphabetical, lower or upper
# and print adequate ciphertext
for c in plain:
    if c.isalpha():
        if c.islower():
            if ord(c) + key > 122:
                temp = (ord(c) - 97 + key) % 26
                print(chr(temp + 97), end = '')
            else:
                temp = ord(c) + key
                print(chr(temp), end = '')
        elif c.isupper():
            if ord(c) + key > 90:
                temp = (ord(c) - 65 + key) % 26
                print(chr(temp + 65), end = '')
            else:
                temp = ord(c) + key
                print(chr(temp), end = '')
    else:
        print(c, end = '')

print()
exit(0)

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    // check if number of arguments is 2 (if not, throw error code 1)
    if(argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    else
    {
        // convert 2nd argument from string to int
        int key = atoi(argv[1]);

        // get plaintext from user
        printf("plaintext: ");
        string s = get_string();

        // iterate plaintext and check if current character is a letter, lower or upper case...
        // ...check if current character + key exceed number of letters in alphabet (if so, do: key % 26)
        printf("ciphertext: ");
        for(int i = 0, l = strlen(s); i < l; i++)
        {
            if(isalpha(s[i]))
            {
                if(islower(s[i]))
                {
                    if(s[i] + key > 122)
                    {
                        printf("%c", (s[i] - 97 + key) % 26 + 97);
                    }
                    else
                    {
                        printf("%c", s[i] + key);
                    }
                }
                else if(isupper(s[i]))
                {
                    if(s[i] + key > 90)
                    {
                        printf("%c", (s[i] - 65 + key) % 26);
                    }
                    else
                    {
                        printf("%c", s[i] + key);
                    }
                }
            }
            else
            {
                printf("%c", s[i]);
            }
        }
        printf("\n");
        return 0;
    }
}